server/src/test/java/org/elasticsearch/transport/RemoteClusterPortSettingsTests.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.transport;
+
+import org.elasticsearch.common.network.NetworkAddress;
+import org.elasticsearch.common.network.NetworkService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.List;
+
+import static org.elasticsearch.transport.RemoteClusterPortSettings.REMOTE_CLUSTER_PORT_ENABLED;
+import static org.elasticsearch.transport.RemoteClusterPortSettings.REMOTE_CLUSTER_PROFILE;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.equalTo;
+
+public class RemoteClusterPortSettingsTests extends ESTestCase {
+    /**
+     * Tests that, if Remote Cluster Security 2.0 is enabled, we reject any configuration of that profile
+     * via the profile settings.
+     */
+    public void testRemoteClusterProfileCannotBeUsedWhenRcs2IsEnabled() {
+        List<Setting.AffixSetting<?>> transportProfileSettings = List.of(
+            TransportSettings.TCP_KEEP_ALIVE_PROFILE,
+            TransportSettings.TCP_KEEP_IDLE_PROFILE,
+            TransportSettings.TCP_KEEP_INTERVAL_PROFILE,
+            TransportSettings.TCP_KEEP_COUNT_PROFILE,
+            TransportSettings.TCP_NO_DELAY_PROFILE,
+            TransportSettings.TCP_REUSE_ADDRESS_PROFILE,
+            TransportSettings.TCP_SEND_BUFFER_SIZE_PROFILE,
+            TransportSettings.TCP_RECEIVE_BUFFER_SIZE_PROFILE,
+            TransportSettings.BIND_HOST_PROFILE,
+            TransportSettings.PUBLISH_HOST_PROFILE,
+            TransportSettings.PORT_PROFILE,
+            TransportSettings.PUBLISH_PORT_PROFILE
+        );
+
+        for (Setting.AffixSetting<?> profileSetting : transportProfileSettings) {
+            Settings testSettings = Settings.builder()
+                .put(REMOTE_CLUSTER_PORT_ENABLED.getKey(), true)
+                // We can just stick a random value in, even if it doesn't match the type - that validation happens at a different layer
+                .put(profileSetting.getConcreteSettingForNamespace(REMOTE_CLUSTER_PROFILE).getKey(), randomAlphaOfLength(5))
+                .build();
+            expectThrows(IllegalArgumentException.class, () -> RemoteClusterPortSettings.validateRemoteAccessSettings(testSettings));
```

Comment:
Tots optional but might also assert on part of the message here (i.e., that it starts with "Remote Access settings should not be configured using the"), given that we are putting in a random, potentially invalid value above.

Reply:
Added the assertion for error message 3d93930b9896cc6732bd4a91ea1dea1426ae2059